src/Common/parseGlobs.cpp

```diff
@@ -116,4 +121,67 @@ std::string makeRegexpPatternFromGlobs(const std::string & initial_str_with_glob
     }
     return buf_final_processing.str();
 }
+
+void expandSelector(const std::string & path, Strings & for_match_paths_expanded)
+{
+    /// regexp for {expr1,expr2,expr3}, expr.. should be without "{", "}", "*" and ","
+    static const re2::RE2 selector_regex(R"({([^{}*,]+,[^{}*]*[^{}*,])})");
+
+    std::string_view path_view(path);
+    std::string_view matched;
+
+    if (RE2::FindAndConsume(&path_view, selector_regex, &matched))
+        std::string buffer(matched);
+    else
+    {
+        for_match_paths_expanded.push_back(path);
+        return;
+    }
+
+    Strings expanded_paths;
```

Comment:
I know the code was moved here from elsewhere but there are lots of subtle changes which make it hard to double-check by just reading it. Please add some comments where appropriate to the code.